Stratis.Bitcoin/Features/MemoryPool/MemPoolCoinView.cs

```diff
@@ -9,35 +9,103 @@
 
 namespace Stratis.Bitcoin.Features.MemoryPool
 {
+    /// <summary>
+    /// Memory pool coin view.
+    /// Provides coin view representation of memory pool transactions via a backed coin view.
+    /// </summary>
     public class MempoolCoinView : CoinView , IBackedCoinView
     {
+        #region Fields
+
+        /// <summary>Transaction memory pool for managing transactions in the memory pool.</summary>
         private readonly TxMempool memPool;
-        private readonly AsyncLock mempoolScheduler;
+
+        /// <summary>A lock for managing asynchronous access to memory pool.</summary>
+        private readonly AsyncLock mempoolLock;
+
+        /// <summary>Memory pool validator for validating transactions.</summary>
         private readonly IMempoolValidator mempoolValidator;
 
-        public UnspentOutputSet Set { get; private set; }
-        public CoinView Inner { get; }
+        #endregion
+
+        #region Constructor
 
-        public MempoolCoinView(CoinView inner, TxMempool memPool, AsyncLock mempoolScheduler, IMempoolValidator mempoolValidator)
+        /// <summary>
+        /// Constructs a memory pool coin view.
+        /// </summary>
+        /// <param name="inner">The backing coin view.</param>
+        /// <param name="memPool">Transaction memory pool for managing transactions in the memory pool.</param>
+        /// <param name="mempoolLock">A lock for managing asynchronous access to memory pool.</param>
+        /// <param name="mempoolValidator">Memory pool validator for validating transactions.</param>
+        public MempoolCoinView(CoinView inner, TxMempool memPool, AsyncLock mempoolLock, IMempoolValidator mempoolValidator)
         {
             this.Inner = inner;
             this.memPool = memPool;
-            this.mempoolScheduler = mempoolScheduler;
+            this.mempoolLock = mempoolLock;
             this.mempoolValidator = mempoolValidator;
             this.Set = new UnspentOutputSet();
         }
 
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Gets the unspent transaction output set.
+        /// </summary>
+        public UnspentOutputSet Set { get; private set; }
```

Comment:
I am not sure we use that rule from SC, because i certainly do put fields and properties together in front of constructors and it does not complain.

@mikedennis are you and me using the same set of SC rules? I have those from @bokobza that are currently in the Stratis repo.